jax/lax/lax.py

```diff
@@ -484,15 +484,22 @@ def conv_general_dilated(lhs, rhs, window_strides, padding, lhs_dilation=None,
   if type(dimension_numbers) is not ConvDimensionNumbers:
     dimension_numbers = conv_dimension_numbers(
         lhs.shape, rhs.shape, dimension_numbers)
-  if isinstance(padding, str):
-    lhs_perm, rhs_perm, _ = dimension_numbers
-    padding = padtype_to_pads(
-        onp.take(lhs.shape, lhs_perm)[2:], onp.take(rhs.shape, rhs_perm)[2:],
-        window_strides, padding)
   if lhs_dilation is None:
     lhs_dilation = (1,) * (lhs.ndim - 2)
+  elif isinstance(padding, str) and not len(lhs_dilation) == lhs_dilation.count(1):
+    warnings.warn(
```

Comment:
Instead of a warning, how about an error? (Also, maybe the message could say more about how to specify the required padding explicitly, i.e. what the alternative to a string is.)

Reply:
Will put an error in and am planning on adding a PR supporting this (probably in conv_transpose) but there may be a slight delay.

For now would a link to a paper reference be sufficient?

Reply:
Have actually added an error so they use conv_transpose for now and will fix that to allow dilation